src/services/codefixes/importFixes.ts

```diff
@@ -169,32 +181,36 @@ namespace ts.codefix {
             for (const exported of checker.getExportsOfModule(moduleSymbol)) {
                 if ((exported.escapedName === InternalSymbolName.Default || exported.name === symbolName) && skipAlias(exported, checker) === exportedSymbol) {
                     const isDefaultExport = checker.tryGetMemberInModuleExports(InternalSymbolName.Default, moduleSymbol) === exported;
-                    result.push({ moduleSymbol, importKind: isDefaultExport ? ImportKind.Default : ImportKind.Named });
+                    result.push({ moduleSymbol, importKind: isDefaultExport ? ImportKind.Default : ImportKind.Named, exportedSymbolIsTypeOnly: isTypeOnlySymbol(exported) });
                 }
             }
         });
         return result;
     }
 
+    function isTypeOnlySymbol(s: Symbol): boolean {
+        return !(s.flags & SymbolFlags.Value);
+    }
+
     function getFixForImport(
         exportInfos: ReadonlyArray<SymbolExportInfo>,
         symbolName: string,
-        symbolToken: Node | undefined,
+        position: number | undefined,
         program: Program,
         sourceFile: SourceFile,
         host: LanguageServiceHost,
         preferences: UserPreferences,
     ): ReadonlyArray<ImportFix> {
         const checker = program.getTypeChecker();

```

Comment:
Why not exit early here when position is undefined ? without doing any of the work.

Reply:
We actually might return a result if we don't need to add a qualifier to the use. We would add an import at the top of the file and not need `position`.